File: yt-message-admin/yt-message-admin-server/src/main/java/com/yt/message/admin/server/pojo/entity/Platform.java
package com.yt.message.admin.server.pojo.entity;

import com.baomidou.mybatisplus.annotation.TableId;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.io.Serializable;

/**
 * <p>
 * 
 * </p>
 *
 * @author yt
 * @since 2024-11-26
 */
@EqualsAndHashCode(callSuper = true)
@Data
public class Platform extends  BaseEntity implements Serializable {

    private static final long serialVersionUID = 1L;

    /**
     * 平台id
     */
    @TableId(value = "platform_id")
    private Long platformId;

    /**
     * 平台名
     */
    private String platformName;

    /**
     * 这条记录对应消息类型
     */
    private Integer messageType;

    /**
    *限流key
    */

    private String rateLimitKey;
    /**
     * 平台服务限流大小
     */
    private Integer rateLimitScale;

    /**
     * 平台服务限流间隔
     */
    private Integer rateLimitInterval;





    /**
     * 应用id
     */
    private String appId;

    /**
     * 访问id
     */
    private String secretId;

    /**
     * 访问key
     */
    private String secretKey;

    /**
     * 签名内容
     */
    private String signName;

    /**
     * 请求服务器的地址
     */
     private String host;

    /**
     * 请求服务器的口
     */
    private Integer port;

    /**
     * 对应的handlerName
     */

    private String handlerName;


    private Integer version;

    /**
     * 状态
     */
    private Integer status;

}

File: yt-message-admin-ui/vue-vben-admin/apps/web-ele/src/api/system/platform.ts
import type { BasePageReqParams, PageRsp } from "../common/types";
import { requestClient } from '#/api/request';
export namespace platformApi {
  export const baseUri = '/platform'

  /** 分页参数 */
  export interface pageParams extends BasePageReqParams {
    messageType: number;
  }
  export interface PlatformRsp {
    platformId: string;
    platformName: string;
    messageType: number;
    rateLimitKey: string;
    rateLimitScale: number;
    rateLimitInterval: number;
    signName: string;
    host: string;
    port: number;
    handlerName: string;
    version:number;
    
  }
}
export async function page(data: platformApi.pageParams) {
  return requestClient.post<PageRsp<platformApi.PlatformRsp>>(platformApi.baseUri + '/page', data);
}
export async function update(data: any) {
  return requestClient.put(platformApi.baseUri, data)
}
export async function add(data: any) {
  return requestClient.post(platformApi.baseUri, data)
}
export async function remove(id : string) {
  return requestClient.delete(platformApi.baseUri + '/' + id)
}
export async function status(data:any) {
  return requestClient.patch(platformApi.baseUri + '/status', data)
  
}
